File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/ProductoDao.vb
Class ProductoDao
    Public Function getAll() As List(Of Producto)
        Dim lst As List(Of Producto) = New List(Of Producto)()
        Dim dm As DataManager = New DataManager()

        Try
            dm.Open()

            For Each row As DataRow In dm.ConsultaSQL("Select * from Productos").Rows
                lst.Add(map(row))
            Next

        Catch ex As Exception
            dm.Rollback()
        Finally
            dm.Close()
        End Try

        Return lst
    End Function

    Private Function map(ByVal row As DataRow) As Producto
        Dim oProducto As Producto = New Producto()
        oProducto.id_Producto = Convert.ToInt32(row("id_producto").ToString())
        oProducto.nombre = row("nombre").ToString()
        Return oProducto
    End Function
End Class

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/C#_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/ProductoDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.DataLayer.DAOs
{
    class ProductoDao
    {
        public List<Producto> getAll()
        {
            List<Producto> lst = new List<Producto>();
            DataManager dm = new DataManager();
            try
            {
                dm.Open();
                foreach (DataRow row in dm.ConsultaSQL("Select * from Productos").Rows)
                    lst.Add(map(row));

            }
            catch (Exception ex)
            {
                dm.Rollback();
            }
            finally
            {
                // Cierra la conexión 
                dm.Close();
            }

            return lst;
        }

        private Producto map(DataRow row)
        {
            Producto oProducto = new Producto();
            oProducto.id_Producto = Convert.ToInt32(row["id_producto"].ToString());
            oProducto.nombre = row["nombre"].ToString();
            return oProducto;
        }
    }
}
